go/libkb/chain_link_v2_test.go

```diff
@@ -110,3 +111,51 @@ func TestOuterLinkV2WithMetadataPointerContainerDecode(t *testing.T) {
 	err = MsgpackDecode(&o, bytes)
 	requireErrorHasSuffix(t, errCodecDecodeSelf, err)
 }
+
+func serdeOuterLink(t *testing.T, hPrevSeqno *keybase1.Seqno, hPrevHash LinkID, addHPrevHash bool) OuterLinkV2 {
+	o := OuterLinkV2{Version: 2, Seqno: keybase1.Seqno(2), Prev: nil, Curr: nil, LinkType: 1, SeqType: 1, IgnoreIfUnsupported: false}
+	if hPrevSeqno != nil {
+		o.HPrevSeqno = hPrevSeqno
+	}
+	if addHPrevHash {
+		o.HPrevHash = hPrevHash
+	}
+
+	s, err := o.Encode()
+	require.NoError(t, err)
+	o2 := OuterLinkV2{}
+	err = MsgpackDecode(&o2, s)
+	require.NoError(t, err)
+	return o2
+}
+
+func hPrevInfoMatch(ol OuterLinkV2, hPrevInfo *HPrevInfo) error {
+	return ol.AssertFields(ol.Version, ol.Seqno, ol.Prev, ol.Curr, ol.LinkType, ol.SeqType, ol.IgnoreIfUnsupported, hPrevInfo)
+}
+
+func TestHPrevBackwardsCompatibility(t *testing.T) {
+	o := serdeOuterLink(t, nil, nil, false)
+	require.True(t, o.HPrevSeqno == nil)
+	require.True(t, o.HPrevHash == nil)
+	require.NoError(t, hPrevInfoMatch(o, nil))
+
+	seqno := keybase1.Seqno(3)
+	o = serdeOuterLink(t, &seqno, nil, false)
```

Comment:
this is kind of a weird test. you're creating a new outer link with a seqno of 2 but that has a previous high sequence number of 3? maybe the serdeOuterLink helper should also take seqno as a parameter so the numbers can at least advance in the same direction.